src/AdvancedADC.h

```diff
@@ -41,21 +43,74 @@ class AdvancedADC {
             for (auto p : {p0, args...}) {
                 adc_pins[n_channels++] = analogPinToPinName(p);
             }
+            dualMode=false;
+            selectedADC=0;
         }
         AdvancedADC(): n_channels(0), descr(nullptr) {}
         ~AdvancedADC();
         bool available();
         SampleBuffer read();
-        int begin(uint32_t resolution, uint32_t sample_rate, size_t n_samples, size_t n_buffers);
-        int begin(uint32_t resolution, uint32_t sample_rate, size_t n_samples, size_t n_buffers, size_t n_pins, pin_size_t *pins) {
+        int begin(uint32_t resolution, uint32_t sample_rate, size_t n_samples, size_t n_buffers, bool do_start=true,uint8_t adcNum=0);
+        int begin(uint32_t resolution, uint32_t sample_rate, size_t n_samples, size_t n_buffers, size_t n_pins, pin_size_t *pins, bool start=true,uint8_t fixed_adc=0) {
             if (n_pins > AN_MAX_ADC_CHANNELS) n_pins = AN_MAX_ADC_CHANNELS;
             for (size_t i = 0; i < n_pins; ++i) {
                 adc_pins[i] = analogPinToPinName(pins[i]);
             }
+
             n_channels = n_pins;
-            return begin(resolution, sample_rate, n_samples, n_buffers);
+            return begin(resolution, sample_rate, n_samples, n_buffers,start,fixed_adc);
         }
+        
+        void clear();
+
+        int start(uint32_t sample_rate);
+        
+        //void setADCDualMode(bool dm);
+        //int enableDualMode();
+        //int disableDualMode();
+        
         int stop();
 };
 
+class AdvancedADCDual {
+		private:
+			size_t n_channels;
+			pin_size_t adc_pins_unmapped[AN_MAX_ADC_CHANNELS];
+			AdvancedADC *adcIN1;
+			AdvancedADC *adcIN2;
```

Comment:
```suggestion
			AdvancedADC &adc1;
			AdvancedADC &adc2;
```
Those should be passed by reference.